### PR TITLE
[Seq] Add a new op interface for clock transformation ops. NFC

### PR DESCRIPTION
**Pull request by prithayan**
Add the clock gate, clock inverter and clock divider ops to a new clock transformation interface.
This makes it easier to handle them and get the input clock directly.

### CODE REVIEW

**Review by uenoku — COMMENTED** [review]
Does `ClockMuxOp` need to have this interface?  In that case `getClk` returning a single value seems not composable.

**Comment by prithayan** [comment]
> Does `ClockMuxOp` need to have this interface? In that case `getClk` returning a single value seems not composable.

Yeah, I was not sure about clock mux, for the use case that I was considering, `ClockMuxOp`  needs to be handled separately, unlike the others .  But it makes sense to include it as a clock transformation op, and let the client deal with it, if it returns multiple input clocks. I will update the interface to include mux ops.

File: include/circt/Dialect/Seq/SeqOpInterfaces.td [hunk 1]
```diff
@@ -24,6 +24,15 @@ def Clocked : OpInterface<"Clocked"> {
   ];
 }
 
+def ClockTransformation : OpInterface<"ClockTransformation"> {
+  let cppNamespace = "circt::seq";
+  let description = "Trait for a clock transformation operation";
+
+  let methods = [
+    InterfaceMethod<"Get the input clock signal", "Value", "getClk", (ins)>
```
Comment (darthscsi) [inline include/circt/Dialect/Seq/SeqOpInterfaces.td:32]: Is a single value here correct?

File: include/circt/Dialect/Seq/SeqOpInterfaces.td [hunk 2]
```diff
@@ -24,6 +24,15 @@ def Clocked : OpInterface<"Clocked"> {
   ];
 }
 
+def ClockTransformation : OpInterface<"ClockTransformation"> {
```
Comment (darthscsi) [inline include/circt/Dialect/Seq/SeqOpInterfaces.td:27]: Do we need a trait or just a function?  Are other dialects getting in on the action?  Are there times we just care about `ClockTransformation` ops but not what they do?

**Comment by prithayan** [comment]
Closing, as this can be accomplished via a simple function, the itnerface is not relevant.

**Pull request closed without merging**